### DISCUSSION TITLE
Route recursion issue with orderable collection

### DISCUSSION THREAD

**Discussion by wetnet** [statamic/cms#6982] [Q&A] [answered]
I have a few orderable collections with routes `{mount}/{parent_uri}/{slug}`.
The problem here is that the deeper I go in the structure, the more {mount} segments are repeated:
`/mount-slug/mount-slug/mount-slug/grandparent-slug/parent-slug/slug`

If I omit the `{parent_uri}` then I always get a single `/mount-slug/` but of course no nested structure either: 
`/mount-slug/slug` (no ancestors)

How can I do this?

**Comment by jasonvarga** [comment, marked as answer]
You can use a `depth` condition in the route, example shown here: https://statamic.dev/collections#organizing-gaming-articles

You can do this:

```yml
route: '{{ depth == 1 ?= mount }}/{{ parent_uri }}/{{ slug }}'
```

**Reply by wetnet** [reply]
Aha! Thank you Jason.
